dvc/output/local.py

```diff
@@ -118,3 +126,17 @@ def save(self):
                 self.repo.scm.ignore(self.path)
 
         self.info = self.remote.save_info(self.path_info)
+
+    def get_dir_cache(self):
+        if not self.dir_cache:
```

Comment:
Let's tie it down to the current checksum, so that if output changes (e.g. `dvc repro` happened) we don't accidentally return old dir cache.